gpkit/__init__.py

```diff
@@ -95,6 +96,50 @@ def __call__(self, arg):
 
 enable_units()
 
+VECTORIZATION = ()
+MODELS = []
+MODELNUMS = []
+MODELNUM_LOOKUP = defaultdict(int)
+
+
+def begin_variable_naming(model):
+    global MODELS, MODELNUMS
+    MODELS.append(model)
+    num = MODELNUM_LOOKUP[model]
+    MODELNUMS.append(num)
+    MODELNUM_LOOKUP[model] += 1
+    return num, (tuple(MODELS), tuple(MODELNUMS))
+
+
+def end_variable_naming():
+    global MODELS, MODELNUMS
+    MODELS = MODELS[:-1]
+    MODELNUMS = MODELNUMS[:-1]
```

Comment:
are `MODELS` and `MODELNUMS` lists, or not? It's confusing that they start out being assigned as `[]` and get appended to by `begin_variable_naming`, but then get assigned to the last element here. What happens if we then enter `begin_variable_naming` again (which will try to append) ?